src/ogma/features/nodeGrouping.ts

```diff
@@ -126,18 +125,17 @@ export class NodeGroupingTransformation {
    * run layout on all subnodes of virtual nodes
    */
   public async runLayoutOnAllSubNodes(): Promise<void> {
-    // @ts-ignore getContext exists on the transformation but hidden by the types
-    const virtualNodes = this.transformation.getContext().virtualNodes;
-    const rawNodesList = virtualNodes.getSubNodes();
+    const rawNodesList = this._getAllTransformationRawNodes();
     const promisesList: Promise<void>[] = [];
     for (let i = 0; i < rawNodesList.length; i++) {
-      const subNodes = rawNodesList[i];
+      // rawNodesList[i] is not null because each group has at least one node
+      const subNodes = rawNodesList[i]!;
       if (subNodes !== undefined) {
```

Comment:
why do you check for undefined since following your comment just above 'rawNodesList[i] is not null because each group has at least one node'

Reply:
The was before the changes, no need for the undefined check